scanpipe/pipes/__init__.py

```diff
@@ -254,27 +260,65 @@ def update_or_create_dependency(
     corresponding CodebaseResource for `datafile_path`. This is used in the case
     where Dependency data is imported from a scancode-toolkit scan, where the
     root path segments are not stripped for `datafile_path`.
+    If the dependency is resolved and a resolved package is created, we have the
+    corresponsing package_uid at `resolved_to`.
     """
     dependency = None
     dependency_uid = dependency_data.get("dependency_uid")
+    extracted_requirement = dependency_data.get("extracted_requirement")
 
     if ignore_dependency_scope(project, dependency_data):
         return  # Do not create the DiscoveredDependency record.
 
+    dependencies = []
     if not dependency_uid:
-        dependency_data["dependency_uid"] = uuid.uuid4()
+        purl_data = DiscoveredDependency.extract_purl_data(dependency_data)
+        dependencies = DiscoveredDependency.objects.filter(
+            project=project,
+            extracted_requirement=extracted_requirement,
+            **purl_data,
+        )
     else:
-        dependency = project.discovereddependencies.get_or_none(
+        dependency = DiscoveredDependency.objects.get_or_none(
+            project=project,
             dependency_uid=dependency_uid,
         )
+        if dependency:
+            dependencies.append(dependency)
+
+    for dependency in dependencies:
+        # This dependency relationship is for a new package
+        if (
+            for_package
+            and dependency.for_package
+            and dependency.for_package != for_package
+        ):
```

Comment:
Instead of a comment, put the condition in a variable with an explicit name, for example, `is_for_new_package`.

Reply:
Added, thanks!